### PR TITLE
test: Fail fast when image build fails on tests #11102

### PR DESCRIPTION
**Pull request by ElayAharoni**
added check in build-images.sh to check if the build has failed then kill the process with error.
fix https://github.com/kubeflow/pipelines/issues/11102

### CODE REVIEW

File: scripts/deploy/github/build-images.sh [hunk 1]
```diff
@@ -20,14 +20,31 @@ set -e
...
 REGISTRY="${REGISTRY:-kind-registry:5000}"
 TAG="${TAG:-latest}"
+EXIT_CODE=0
 
 docker system prune -a -f
 
-docker build -q -t "${REGISTRY}/apiserver:${TAG}" -f backend/Dockerfile . && docker push "${REGISTRY}/apiserver:${TAG}" &
-docker build -q -t "${REGISTRY}/persistenceagent:${TAG}" -f backend/Dockerfile.persistenceagent . && docker push "${REGISTRY}/persistenceagent:${TAG}" &
-docker build -q -t "${REGISTRY}/scheduledworkflow:${TAG}" -f backend/Dockerfile.scheduledworkflow . && docker push "${REGISTRY}/scheduledworkflow:${TAG}" &
+dacker build -q -t "${REGISTRY}/apiserver:${TAG}" -f backend/Dockerfile . && docker push "${REGISTRY}/apiserver:${TAG}" || EXIT_CODE=$?
```
Comment (rimolive) [inline scripts/deploy/github/build-images.sh:27]: s/dacker/docker/

**Review by rimolive — CHANGES_REQUESTED** [review]

**Comment by ElayAharoni** [comment]
i am trying to see the failing workflow now this is why i break the DockerFile.
/rerun-all

File: scripts/deploy/github/build-images.sh (hunk 1 above)
Comment (ElayAharoni) [reply scripts/deploy/github/build-images.sh:27]: i did that in order for the action to fail so i could see the issue, i changed it back and broke the DockerFile instead

**Comment by ElayAharoni** [comment]
here is a recording of the test when the image build is failing (broken DockerFile).
[video of failing test](https://www.loom.com/share/d881029f3f2d4a86be639019973534fb?sid=1a891843-4855-4e6a-8103-6d4d2b57d526)
as you can see here when the build is failing the workflow fails instantly with an informative message.
image

**Comment by ElayAharoni** [comment]
i have pushed the fix now without the broken DockerFile.
please review when possible.
/rerun-all

**Comment by rimolive** [comment]
/lgtm

cc @DharmitD @hbelmiro @chensun @zijianjoy

**Review by hbelmiro — COMMENTED** [review]
/lgtm

Thank you for working on this @ElayAharoni.

**Review by DharmitD — COMMENTED** [review]
/Approve

Thank you @ElayAharoni !

**Pull request merged**